File: server/internal/domain/base/entity/dict.go
package entity

import "time"

// Dict 字典领域实体
// 纯业务模型，无 GORM 标签
type Dict struct {
	ID          int64      // 字典 ID
	DictName    string     // 字典名称
	DictCode    string     // 字典编码
	Description string     // 描述
	Status      int        // 状态：1-启用 0-禁用
	DictItems   []DictItem // 字典项列表（一对多关系）
	CreatedBy   int64      // 创建人 ID
	CreatedAt   time.Time  // 创建时间
	UpdatedBy   int64      // 更新人 ID
	UpdatedAt   time.Time  // 更新时间
}

// IsActive 检查字典是否启用
func (d *Dict) IsActive() bool {
	return d.Status == 1
}

// DictItem 字典项领域实体
// 纯业务模型，无 GORM 标签
type DictItem struct {
	ID          int64     // 字典项 ID
	DictID      int64     // 字典 ID
	ItemKey     string    // 字典项键
	ItemValue   string    // 字典项值
	Sort        int       // 排序
	Description string    // 描述
	Status      int       // 状态：1-启用 0-禁用
	CreatedBy   int64     // 创建人 ID
	CreatedAt   time.Time // 创建时间
	UpdatedBy   int64     // 更新人 ID
	UpdatedAt   time.Time // 更新时间
}

// IsActive 检查字典项是否启用
func (di *DictItem) IsActive() bool {
	return di.Status == 1
}

File: web/src/types/dict.ts
// 字典相关类型定义
// 所有 ID 字段使用 string 类型，避免 JavaScript Number 精度丢失问题

// 字典项信息
export interface DictItem {
  id: string // 字典项 ID
  dictId: string // 字典 ID
  itemKey: string // 字典项键
  itemValue: string // 字典项值
  sort?: number // 排序
  description?: string // 描述
  status?: number // 状态：1-启用 0-禁用
  createdAt?: string // 创建时间
  updatedAt?: string // 更新时间
}

// 字典信息
export interface Dict {
  id: string // 字典 ID
  dictName: string // 字典名称
  dictCode: string // 字典编码
  description?: string // 描述
  status?: number // 状态：1-启用 0-禁用
  createdAt?: string // 创建时间
  updatedAt?: string // 更新时间
  dictItems?: DictItem[] // 字典项列表
}

// 字典列表响应
export interface DictListResponse {
  list: Dict[]
  total: number
  page: number
  pageSize: number
}

// 字典项列表响应
export interface DictItemListResponse {
  list: DictItem[]
  total: number
  page: number
  pageSize: number
}

// 创建字典请求
export interface CreateDictRequest {
  dictName: string
  dictCode: string
  description?: string
  status?: number
}

// 更新字典请求
export interface UpdateDictRequest {
  id: string
  dictName?: string
  dictCode?: string
  description?: string
  status?: number
}

// 创建字典项请求
export interface CreateDictItemRequest {
  dictId: string
  itemKey: string
  itemValue: string
  sort?: number
  description?: string
  status?: number
}

// 更新字典项请求
export interface UpdateDictItemRequest {
  id: string
  itemKey?: string
  itemValue?: string
  sort?: number
  description?: string
  status?: number
}
